src/authenticatedFetch/dpop/DpopAuthenticatedFetcher.ts

```diff
@@ -75,16 +86,14 @@ export default class DpopAuthenticatedFetcher implements IAuthenticatedFetcher {
       // perform unauthenticated fetch
       return this.fetcher.fetch(url, {
         ...requestInit,
-        headers: {
-          ...requestInitiWithDefaults.headers
-        }
+        headers: flattenHeaders(requestInitiWithDefaults?.headers)
```

Comment:
Can be replaced by `Object.fromEntries(new Headers(requestInitiWithDefaults?.headers).toEntries())`

Reply:
Or alternatively, change the `flattenHeaders` method to that implementation for easier reuse.

Reply:
TypeScript complains that there is no `toEntries` or `entries` methods in the `Headers` interface, even if it is mandated by the [MDN spec](https://developer.mozilla.org/en-US/docs/Web/API/Headers). Here is the `Headers` interdaces as per `lib.dom.d.ts`:
```
/** This Fetch API interface allows you to perform various actions on HTTP request and response headers. These actions include retrieving, setting, adding to, and removing. A Headers object has an associated header list, which is initially empty and consists of zero or more name and value pairs.  You can add to this using methods like append() (see Examples.) In all methods of this interface, header names are matched by case-insensitive byte sequence. */
interface Headers {
    append(name: string, value: string): void;
    delete(name: string): void;
    get(name: string): string | null;
    has(name: string): boolean;
    set(name: string, value: string): void;
    forEach(callbackfn: (value: string, key: string, parent: Headers) => void, thisArg?: any): void;
}
```

Similar mismatches might very well be causing the observed issue in the first place.

Reply:
Mwm, we should just cast here then; TypeScript is wrong.

Reply:
The types definition in `node-fetch` seem to be aligned with the spec, I'll use them for casting.

Reply:
Weird that they don't accept `Headers` then though.

Reply:
Might be because MDN still lists `entries` as:

>     Experimental. Expect behavior to change in the future.

Otherwise, consider [contributing it upstream](https://github.com/Microsoft/TypeScript/blob/master/CONTRIBUTING.md#contributing-libdts-fixes).

Reply:
To be fair, all properties of the object are experimental 😁

Reply:
Actually casting this would have an impact on several interfaces, such as `IFetcher` and `IAuthenticatedFetcher`, and I think it will have rippling effects all over the place. Since the proposed fix patches the issue at hand right now, I suggest that a deeper typing refactoring is looked at in a different PR. If you think it's an appropriate approach, I'll create the associated issue.

Reply:
Was thinking about a quick local `any` cast for `entries`, but no strong opinion.

Reply:
That's... A good point :P Then maybe consider a PR to TS :)